Source/iop/Iop_Sysclib.cpp

```diff
@@ -241,6 +248,16 @@ void CSysclib::Invoke(CMIPS& context, unsigned int functionId)
 			context.m_State.nGPR[CMIPS::V0].nD0 = context.m_State.nGPR[CMIPS::A0].nV0;
 		}
 		break;
+	case 42:
+		// int vsprintf (char * s, const char * format, va_list arg );
+		{
+			context.m_State.nGPR[CMIPS::V0].nD0 = static_cast<int>(__vsprintf(
+				reinterpret_cast<char*>(&m_ram[context.m_State.nGPR[CMIPS::A0].nV0]),
+				reinterpret_cast<const char*>(&m_ram[context.m_State.nGPR[CMIPS::A1].nV0]),
+				reinterpret_cast<va_list>(&m_ram[context.m_State.nGPR[CMIPS::A2].nV0])

```

Comment:
yes, you have to do it this way because va_list is passed to you. va_start initialises a va_list by taking the argument before the elipsis and setting the va_list value to the correctly aligned pointer. We're passed va_list (which is just a pointer) so we don't want to initialise it. va_end just sets the va_list pointer to null so you cannot accidently use it again. The documentation for va_* isn't really that clear.